sql/core/src/main/scala/org/apache/spark/sql/DataFrameWriter.scala

```diff
@@ -323,18 +323,25 @@ final class DataFrameWriter[T] private[sql](ds: Dataset[T]) {
           provider match {
             case supportsExtract: SupportsCatalogOptions =>
               val ident = supportsExtract.extractIdentifier(dsOptions)
-              val catalog = CatalogV2Util.getTableProviderCatalog(
```

Comment:
what's wrong with keep using `CatalogV2Util.getTableProviderCatalog`?

Reply:
I do not know how to get `namespaces` from that `catalog` so i think we could get `namespaces` from `dsOptions` or `ident.namespaces`

Reply:
Why do we want to get namespace from catalog? IIUC we only need the catalog name here, which is simply `catalog.name`

Reply:
thanks! I need to learn more :D

Reply:
`CatalogV2Util.getTableProviderCatalog.name` will return `spark_catalog` when we save default catalog, that is cause of test fail `SupportsCatalogOptionsSuite`. which we need here is `default` namespace got from `ident.namespaces`.

Reply:
WDYT if I revert to
```scala
val catalog = if (ident.namespace().isEmpty) {
  Array(dsOptions.get("catalog"))
} else {
  ident.namespace()
}
```
Thanks!

Reply:
Sry, I kept `val catalog = CatalogV2Util.getTableProviderCatalog` and pushed fix commit